internal/pkg/proc/proc.go

```diff
@@ -0,0 +1,83 @@
+package proc
+
+import (
+	"fmt"
+	"strings"
+
+	"code.cloudfoundry.org/bytefmt"
+	"github.com/prometheus/procfs"
+)
+
+type ProcessList struct {
+	Pid            int
+	PPID           int
+	NumThreads     int
+	VirtualMemory  uint64
+	ResidentMemory uint64
+	CPUTime        float64
+	State          string
+	Command        string
+	Executable     string
+	Args           string
+}
+
+// List processes the list of running processes and
+func List() ([]ProcessList, error) {
+	p, err := procfs.AllProcs()
+	if err != nil {
+		return nil, err
+	}
+
+	return format(p)
+}
+
+func format(procs procfs.Procs) ([]ProcessList, error) {
+	pl := make([]ProcessList, 0, len(procs))
+	var (
+		cmdline []string
+		comm    string
+		err     error
+		exe     string
+		stats   procfs.ProcStat
+	)
+
+	for _, proc := range procs {
+		comm, err = proc.Comm()
+		if err != nil {
+			return nil, err
+		}
+		exe, err = proc.Executable()
+		if err != nil {
+			return nil, err
+		}
+		cmdline, err = proc.CmdLine()
+		if err != nil {
+			return nil, err
+		}
+		stats, err = proc.NewStat()
+		if err != nil {
+			return nil, err
+		}
+
+		p := ProcessList{
+			Pid:            proc.PID,
+			PPID:           stats.PPID,
+			State:          stats.State,
+			NumThreads:     stats.NumThreads,
+			CPUTime:        stats.CPUTime(),
+			VirtualMemory:  uint64(stats.VirtualMemory()),
+			ResidentMemory: uint64(stats.ResidentMemory()),
+			Command:        comm,
+			Executable:     exe,
+			Args:           strings.Join(cmdline, " "),
+		}
+
+		pl = append(pl, p)
+	}
+
+	return pl, nil
+}
+
+func (p *ProcessList) String() string {
```

Comment:
Main reason this is here is for the list widget for the UI since it expects `[]string`.